### ISSUE TITLE
ABSwitch should be a 'switch' instead of toggle button

### ISSUE THREAD

**Issue by jessegreenberg** [phetsims/balloons-and-static-electricity#149]
From #148, @terracoda suggested that we try replacing the ABSwitch toggle button with an accessible switch.

From #148, it should be something like this:
```html
<input id="abswitch-17-61-513-564-557-544-539-543" type="checkbox" role="switch" name="Two-balloon experiement" aria-checked="false">
```

Also, @terracoda pointed out that
> we have to keep in mind that standard screen readers might not recognize this role yet. It's brand new.

**Comment by jessegreenberg** [comment]
@terracoda also mentioned that 

> "Two-balloon experiment, switch, off" which patterns "Two-balloon experiment, toggle button, not pressed". The language for a switch control is much clearer.

So the prototype screen reader should say something like the above for the accessible switch.

**Comment by jessegreenberg** [comment]
@terracoda, I just tested out 

```html
<input id="abswitch-17-61-513-564-557-544-539-543" type="checkbox" role="switch" name="Two-balloon experiement" aria-checked="false">
```
NVDA treats it very similar toggle button with 'aria-pressed'.  The output from NVDA in both Firefox and Chrome is 
"Two Balloon Experiment Button, Toggle Button, not pressed". (when aria-checked=false)
"Two Balloon Experiment Button, Toggle Button, checked, not pressed" (when aria-checked=true)

And it reads "Checked" when it gets checked. Strange that it still labels it as a toggle button and reads 'not pressed' incorrectly.

**Comment by jessegreenberg** [comment]
Jaws is very similar.  Sometimes it labels it as a 'checkbox', others as a 'toggle button'.  The output is

With 'aria-checked'='false': "Two Balloon Experiment Checkbox, Not Checked"
With 'aria-checked='true': "Two Balloon Experiment, Checkbox, Checked"

But when the checkbox is pressed, JAWS reports:
"Toggle Button Checked, Two Balloon Experiment" (When Checked)
"Toggle Button, Two Balloon Experiment" (When un-checked)

**Comment by jessegreenberg** [comment]
Commit above replaces the toggle button with an accessible switch.

**Comment by jessegreenberg** [comment]
Prototype reader now supports an accessible switch.  It seems that Safari does not put Check boxes in the natural navigation order?  We can force it with tabIndex=0.

**Comment by jessegreenberg** [comment]
Setting tabIndex doesn't seem to help.  The checkboxes in this page work well in Safari:
http://webaim.org/techniques/forms/controls

**Comment by jessegreenberg** [comment]
From https://github.com/phetsims/john-travoltage/issues/89#issuecomment-220352858, it was discovered that adding some width to the checkbox fixes the problem in Safari!

**Comment by jessegreenberg** [comment]
Here is a dev version with the switch implementation, and the prototype screen reader behaving like https://github.com/phetsims/balloons-and-static-electricity/issues/149#issuecomment-222246217

http://www.colorado.edu/physics/phet/dev/html/balloons-and-static-electricity/1.2.0-reader.4/balloons-and-static-electricity_en.html?reader

**Comment by terracoda** [comment]
@jessegreenberg Testing results for Mac OSX, El Capitan, Safari 9.1.1 (latest) and VoiceOver.
When I use the VO cursor keys to navigate to the switch, I hear:
- Two-balloon experiment, off, Switch. You are currently on a switch. To select or deselect this checkbox, press Control-Option-Space.

Then when I activate the switch with the VO-commands (or with the spacebar alone) and I hear:
- on, Two balloon experiment, switch

And then another activation:
- off, Two balloon experiment, switch
VO also provides a little click sound on the switch, an earcon the Wall button, which is a simple button, does not have (interesting)!

I think VO's verabalizations with Safari are pretty good. The verbalizations are the same whether I use the native VO-Commands or the spacebar alone to activate the switch (not the case for the Wall button). The Enter key does not activate the switch, and this is the correct native behaviour as it now a checkbox instead of a button.

**Comment by terracoda** [comment]
Testing with the reader, the switch only gains focus if I tab to it. If I use the arrow keys to navigate to it, I cannot activate it by hitting the spacebar. This is different from the Wall button which is operable whether I navigate to it with the tab key or the down arrow.

Oh, and note that the interaction alert, of "Green balloon added or removed from play area" does not fire with VO and Safari, but if course works nicely with the Reader.

**Comment by terracoda** [comment]
And the verbalizations for the switch itself are nice as well:

- Two balloon experiment, switch, Off (with a single balloon)
- Two balloon experiment, switch, On (with two balloons)

**Comment by terracoda** [comment]
@jessegreenberg, by the you have done a great job in implementing general Hotkeys based on structural HTML elements.

**Comment by terracoda** [comment]
@jessegreenberg, while testing with VoiceOver (El Capitan and Safari), I hear an extra text element, "Two balloon experiment". It comes in between the switch and the the switch's description. Can you check to make sure there isn't an extra element in there?

**Comment by jessegreenberg** [comment]
> Testing with the reader, the switch only gains focus if I tab to it.

Thanks for finding this @terracoda, I will take a look.

> Testing results for Mac OSX, El Capitan, Safari 9.1.1 (latest) and VoiceOver.

Thanks for testing!  Happy to hear that VO supports the switch role!

> https://github.com/phetsims/balloons-and-static-electricity/issues/149#issuecomment-222336043

Thanks @terracoda!

> https://github.com/phetsims/balloons-and-static-electricity/issues/149#issuecomment-222336402

Yes, I see an extra element, thanks.  I will remove it.

**Comment by jessegreenberg** [comment]
Focus issue fixed via https://github.com/phetsims/scenery/commit/589de16a419133bf5ff5fab09042e50a08bf19fe

**Comment by jessegreenberg** [comment]
> Yes, I see an extra element, thanks. I will remove it.

Actually, the reader was reading label content twice.  Should be fixed, but I could not reproduce on Windows Chrome.

**Comment by jessegreenberg** [comment]
I think this issue can be closed now that all issues are resolved.  Any new issues can be created after review of a new dev version.

Keeping the switch for now, it works really well with the prototype and VO, and works OK with NVDA and JAWS.
